cdap-ui/app/services/app-level-loading-service.js

```diff
@@ -0,0 +1,25 @@
+angular.module(PKG.name + '.services')
+  .service('myLoadingService', function($q, $rootScope) {
+    var deferred;
+    this.showLoadingIcon = function() {
+      if (deferred) {
+        return deferred.promise;
+      } else {
+        deferred = $q.defer();
+        $rootScope.$broadcast('showLoadingIcon');
```

Comment:
Since this is an app-level loading icon modal I am broadcasting the event on the rootscope. Since all scopes in angular inherit from rootscope this would work. Not sure what EventPipe here means. We need to communicate to all the scopes saying that the loading icon is shown/hidden in this case.